### DISCUSSION TITLE
blastx error - perhaps realted to sequence size

### DISCUSSION THREAD

**Discussion by o-william-white** [bbuchfink/diamond#837] [Q&A] [answered]
Hello, 

I am trying to run a diamond blastx search against the NCBI nr databse for a very large genome (approximately 48Gb). 

I am getting the error below and I haven't been able to find any further explanations for the cause. 

```
Error: vector::_M_range_insert
```

In the meantime, I have rerun the command with `--verbose` to see if that gives further details. 

I was just wondering, **is there any size limit for a diamond blastx search**? Below is is the seqkit stats output for the genome I am trying to blast if helpful. 

```
format  type   num_seqs         sum_len  min_len    avg_len        max_len      Q1       Q2       Q3  sum_gap            N50  Q20(%)  Q30(%)
FASTA   DNA      88,969  48,011,215,492    9,293  539,639.8  4,374,437,920  94,497  121,399  157,622        0  1,082,033,919       0       0
```

**Comment by bbuchfink** [comment, marked as answer]
Diamond is not suitable to process full chromosomes (except microbial) in blastx mode. The only way to do it would be to chop it into smaller pieces.

**Reply by o-william-white** [reply]
Thanks for highlighiting this. I was trying to follow a blobtoolkit analysis, which included a blast or diamond search. I will try to split up the sequences to run diamond.
